### PR TITLE
Add metrics for caas resources

### PR DESCRIPTION
**Pull request by JohnGarbutt**

### CODE REVIEW

File: charts/operator/Chart.yaml [hunk 1]
```diff
@@ -10,3 +10,9 @@ dependencies:
   - name: ara
     version: ">=0-0"
     repository: file://../ara
+  - name: kube-state-metrics
+    repository: https://prometheus-community.github.io/helm-charts
+    version: 5.8.0
+    version: 5.15.2
```
Comment (mkjpryor) [inline charts/operator/Chart.yaml:16]: Two versions

File: charts/operator/values.yaml [hunk 2]
```diff
@@ -39,3 +39,75 @@ affinity: {}
...
+      spec:
+        resources:
+          - groupVersionKind:
+              group: caas.azimuth.stackhpc.com
+              version: v1alpha1
+              kind: Clusters
+            metricNamePrefix: azimuth_caas_clusters
+            labelsFromPath:
+              realm_namespace: [metadata, namespace]
+              realm_name: [metadata, name]
```
Comment (mkjpryor) [inline charts/operator/values.yaml:88]: These should probably be `cluster_name{space}`?

File: charts/operator/values.yaml [hunk 3]
```diff
@@ -39,3 +39,75 @@ affinity: {}
...
+            metrics:
+              - name: phase
+                help: "Cluster phase"
+                each:
+                  type: Info
+                  info:
+                    labelsFromPath:
+                      phase: [status, phase]
```
Comment (mkjpryor) [inline charts/operator/values.yaml:96]: Are there other metrics you want for the clusters? It would be nice to do things like number of clusters by type?

**Review by mkjpryor — CHANGES_REQUESTED** [review]

File: charts/operator/values.yaml (hunk 3 above)
Comment (JohnGarbutt) [reply charts/operator/values.yaml:96]: yeah, I just went simple to start with, but group by type is a good idea

**Review by mkjpryor — APPROVED** [review]
LGTM. We can add the alerting rules and dashboards later, but let's start collecting metrics now.

**Pull request merged**